### PR TITLE
feat: add option to query members and their groups

### PR DESCRIPTION
**Pull request by joaoviana**
Closes: N/A

### Description:

Allows `includeGroups` in query for `/org/users`;

Gets users and the groups they belong to;

> [!NOTE]
> Should we do extra checks on permissions?

Example
Screenshot 2023-12-29 at 16 50 19
Screenshot 2023-12-29 at 16 50 33

### Reviewer actions

- [ ] I have manually tested the changes in the preview environment
- [ ] I have reviewed the code
- [ ] I understand that "request changes" will block this PR from merging

### CODE REVIEW

**Review by magnew — APPROVED** [review]
Looks good. I had some questions for my own learning. I tested this and I think it does what we need.

icludeGroups can be any value? We might want to consider making it either a boolean or a number. The groups version of this takes a number so that we can paginate it. I don't expect there to be as many groups as users, but maybe we should just do the same here. Up to you if you wanna add that here or add it in a future review.

File: packages/backend/src/models/OrganizationMemberProfileModel.ts [hunk 1]
```diff
@@ -31,6 +34,7 @@ type DbOrganizationMemberProfile = {
 
 const SelectColumns = [
     `${UserTableName}.user_uuid`,
+    `${UserTableName}.user_id`,
```
[the remark this answers is not in the document]
Comment (joaoviana) [reply packages/backend/src/models/OrganizationMemberProfileModel.ts:37]: `user_id` is the `user_id` in the organization! the `uuid` is a general uuid!

File: packages/backend/src/models/OrganizationMemberProfileModel.ts [hunk 2]
```diff
@@ -115,6 +119,94 @@ export class OrganizationMemberProfileModel {
...
 
+    async getOrganizationMembersAndGroups(
+        organizationUuid: string,
+    ): Promise<OrganizationMemberProfileWithGroups[]> {
+        const queryResult: (DbOrganizationMemberProfile & {
+            group_uuids: string[];
+            group_names: string[];
+            groups: { name: string; uuid: string }[];
+        })[] = await this.database(UserTableName)
+            .leftJoin(
```
[the remark this answers is not in the document]
Comment (joaoviana) [reply packages/backend/src/models/OrganizationMemberProfileModel.ts:130]: I guess so? 

I just did it like this to retrieve all the columns we needed 😓 
https://github.com/lightdash/lightdash/blob/feat/add-option-to-query-users-and-the-groups-theyre-in/packages/backend/src/models/SavedChartModel.ts#L554

**Comment by joaoviana** [comment]
> Looks good. I had some questions for my own learning. I tested this and I think it does what we need.
> 
> icludeGroups can be any value? We might want to consider making it either a boolean or a number. The groups version of this takes a number so that we can paginate it. I don't expect there to be as many groups as users, but maybe we should just do the same here. Up to you if you wanna add that here or add it in a future review.


> but maybe we should just do the same here.

sure, great point! I've implemented the `includeGroups` to be the same implementation as `includeMembers` :+1:

**Pull request merged**